[FEATURE] Updated TopWinesSection

AnnaRozenbachs opened the pull request:
# [FEATURE] Updated TopWinesSection

## Description

I have updated the component to fetch the real data from `PreSaleWinesContext`, I also fixed styling for desktop and mobileversion (as it is now in figma, don't know if mobileversion is whole finnished yet). I haven't added the ratings, reviews and favourite because `WineCard` component doesn't have it yet. 

## Changes Made

- **UI Components:** `TopWinesSection.js`, `Button.js`, `WineCard.js`, 
- **Styling:** `button.css`
- **Translations:** `en.json`, `dk.json`

## Screenshots (if applicable)

<img width="1061" alt="image" src="https://github.com/user-attachments/assets/87b41acd-6a79-4515-a53d-2b32221e91c0" />
<img width="287" alt="image" src="https://github.com/user-attachments/assets/40259421-05bd-49a0-9f6d-639b26b5a32b" />


## Testing Steps

**Bransch**
feature/update-TopWinesSection

1. Go to home page.
2. Scroll down to `TopWinesSection`. Navigate next/prev button and see if it is as expected
3. Do the same thing for mobileversion. 

## Issues Addressed

Closes https://github.com/Donna-Vino-Aps/donna-vino-ecommerce/issues/213

SimonFromSweden reviewed (changes requested):
The slider now functions nicely. However, there are a few things that need fixing.

- The clickable area of each card now extends outside of the card, which lets the user navigate to a card by clicking outside of it (on the text for example). Try to limit the "clickable area" to only the card
- The positioning of the "see full catalog"-button does not align with the figma design. This might have something to do with clickable area of the cards in the slider pushing it down (?)
- The margins and padding of the text and the buttons in mobile version seem slightly off from Figma design. 
- Some of the font-sizes should change for the mobile version (like the title should change from displayLarge to displayMedium for mobile)
- If I inspect the page and look through certain mobile phones (like Galaxy Z fold 5 or Samsung Galaxy S8+) the positioning of the previous and next buttons look a bit weird and are half-hidden by the component below

AnastasiaSkachenko reviewed (commented):
I checked how top wines section looks now and it is nearly perfect now. There is a small issue, but I am not sure how important this is, but I thought to bring it to your attention any way. 
<img width="830" height="700" alt="Screenshot 2025-07-12 at 19 44 30" src="https://github.com/user-attachments/assets/9f3aada4-87c5-4af9-a12b-4ad2d6374812" />
 on some md screen cards width is not same, for some reason some cards stretch.

SimonFromSweden reviewed (changes requested):
I can see that you have taken the review comments into consideration, and implemented many of the suggested improvements. Now the cards are equal in width, and you have taken all screen sizes into consideration regarding text placement, sizing and more. The one thing that I think needs some fixing right now is for screen sizes below md. When below md (and close to it), and there is only 1 card in the grid, I think the cards look too wide (see image). I think limiting the card size between around 516px and md-breakpoint (768px) is the final improvement that is needed to make the TopWinesSection ready for merging.

<img width="703" height="721" alt="wide-card" src="https://github.com/user-attachments/assets/0bb27589-310e-4ffc-bde0-016f9d8edfd9" />

AnnaRozenbachs commented:
Now I have fixed so we are only fetching top wines data. I also updated the styling for single wine card when you resizing window so I think this should be a better solution than before.

halynabondar commented on src/components/Slider/TopWinesSection.js:
```diff
@@ -5,110 +5,92 @@ import Link from "next/link";
-              <div className="group h-full overflow-visible transition-all duration-300">
+          {wines.map((wine, index) => (
+            <SwiperSlide
+              key={index}
```
Using just the `index` as a key might not be reliable, especially in larger VDOM trees where other elements may also use numeric keys.
It’s a good idea to add some context to make the key more unique and meaningful, for example: `My-Slide-${index}`.

SimonFromSweden reviewed (dismissed):
I see you have done a lot of improvements. The cards now look good when there is only 1 column, the "shadow height" problem is resolved, and you now actually fetch the topwines. Nice work, approved!

halynabondar reviewed (commented):
Great job on this PR.
I’ve added a few small comments!
